### PR TITLE
fix: composer only send message once if user send it twice

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/composer/composer.js [hunk 1]
```diff
@@ -47,6 +48,9 @@ export class Composer extends Component {
     onKeydown(ev) {
         if (ev.key === "Enter") {
             ev.preventDefault(); // to prevent useless return
+            if (!this.state.active) {
+                return;
+            }
```
Comment (seb-odoo) [inline addons/mail/static/src/new/composer/composer.js]: Why is it handled here on top of being handled inside `processMessage`?

File: addons/mail/static/tests/new/composer/composer_tests.js [hunk 2]
```diff
@@ -44,4 +46,34 @@ QUnit.module("mail", (hooks) => {
             assert.containsOnce(target, ".o-mail-emoji-picker");
         }
     );
+
+    QUnit.test("can post a message", async (assert) => {
```
Comment (seb-odoo) [inline addons/mail/static/tests/new/composer/composer_tests.js]: test title looks too generic if the goal is to test non-send of duplicate message

**Review by seb-odoo — COMMENTED** [review]
Ok to merge as it is, but few remarks.

It would have been nice to split commit one for the refactoring of the process, and one for the actual fix.

Also it would be nice if the commit title followed odoo guidelines to avoid breaking ci/style.

**Comment by alexkuhn** [comment]
👆rebased with resolved conflicts
Re-enabled old test that was similar instead of introducing the new one.

**Pull request merged**